include/boost/asio/associated_allocator.hpp

```diff
@@ -25,6 +25,8 @@ namespace boost {
 namespace asio {
 namespace detail {
 
+struct none { BOOST_ASIO_CONSTEXPR none() = default; };
```

Comment:
Even if doing this for MSVC2015 users (as I point out the bogus warning was removed in MSVC2017 or later), there's no need to create such a `none` type. Just use `allocator<char>` instead of `allocator<void>`.